coordinator/shard_mapping.go

```diff
@@ -0,0 +1,232 @@
+package coordinator
+
+import (
+	"runtime"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/influxdata/influxdb/influxql"
+)
+
+type Source struct {
+	Database        string
+	RetentionPolicy string
+}
+
+type ShardInfo struct {
+	Shard        Shard
+	Measurements []string
```

Comment:
Can you add some docs about what these fields are?  Is `Measurements` the subset of the measurements in `Shard` that need to be queried against?